doc/getting_started.rst

```diff
@@ -0,0 +1,185 @@
+
+  >>> from sklearn.ensemble import RandomForestClassifier
+  >>> clf = RandomForestClassifier(random_state=0)
+  >>> X = [[1, 2, 3], [11, 12, 13]]  # 2 samples, 3 features
+  >>> y = [0, 1]  # classes of each sample
+  >>> clf.fit(X, y)
+  RandomForestClassifier(random_state=0)
+
+The :term:`fit` method typically accepts 2 inputs:
+
+- The samples matrix (or design matrix) :term:`X`. The size of ``X``
+  is ``(n_samples, n_features)``, which means that samples are represented
+  as rows and features are represented as columns.
+- The target values :term:`y` which are real number for regression tasks, or
+  integer for classification (they can also be strings). For unsupervized
+  learning tasks, ``y`` need not be specified. ``y`` is usually  1d array where
+  the ith entry corresponds to the target of the ith sample (row) of ``X``.
+
+Both ``X`` and ``y`` are usually expected to be numpy arrays, though other
+formats are also sometimes accepted (e.g. sparse matrices and the more
+general :term:`array-like`).
+
+Once the estimator is fitted, it can be used for predicting target values of
+new data. You don't need to re-train the estimator::
+
+  >>> clf.predict(X)  # predict classes of the training data
+  array([0, 1])
+  >>> clf.predict([[4, 5, 6], [14, 15, 16]])  # predict classes of new data
+  array([0, 1])
+
+Transformers and Pipelines
+--------------------------
+
+Machine learning models are often composed of different parts. A typical
+model consists of a pre-processing step that transforms the data, and a
+final predictor that predicts target values.
+
+In scikit-learn, pre-processors and transformers follow the same API as the
+estimators objects (they actually all inherit from the same
+``BaseEstimator`` class). These objects don't have a :term:`predict` method
+but rather a :term:`transform` method that outputs a newly transformed
+sample matrix ``X``::
+
+  >>> from sklearn.preprocessing import StandardScaler
+  >>> X = [[0, 15],
+  ...      [1, -10]]
+  >>> StandardScaler().fit(X).transform(X)
+  array([[-1.,  1.],
+         [ 1., -1.]])
+
+Transformers and estimators (predictors) can be combined together into a single
+unifying object: a :class:`~sklearn.pipeline.Pipeline`.
+
+In the following example, we :ref:`load the Iris dataset <datasets>`, split it
+into train and test sets, and compute its accuracy score on the test data. The
+pipeline offers the same API as a regular estimator: it can be fitted and
+used for predictions with ``fit`` and ``predict``::
+
+  >>> from sklearn.preprocessing import StandardScaler
+  >>> from sklearn.linear_model import LogisticRegression
+  >>> from sklearn.pipeline import make_pipeline
+  >>> from sklearn.datasets import load_iris
+  >>> from sklearn.model_selection import train_test_split
+  >>> from sklearn.metrics import accuracy_score
+  ...
+  >>> preprocessor = StandardScaler()
+  >>> predictor = LogisticRegression(random_state=0)
+  >>> pipe = make_pipeline(preprocessor, predictor)
```

Comment:
we don't really need the `preprocessor` and `predictor` variables here, and I think in reality we don't encourage people to have them. This tutorial should be what we want people to actually do, since they're gonna be copy/pasting from here.

Reply:
I wanted to make clear that we want instances here, but I agree that can give them bad ideas!